include/nova_renderer/nova_renderer.hpp

```diff
@@ -40,13 +42,34 @@ namespace nova::renderer {
          */
         rhi::Buffer* per_renderable_data = nullptr;
 
-        std::vector<RenderableType> renderables;
+        std::vector<RenderCommandType> commands;
+    };
+
+    template<typename RenderCommandType>
+    struct ProceduralMeshBatch {
+        MapAccessor<MeshId, ProceduralMesh> mesh;
+        
+        /*!
+         * \brief A buffer to hold all the per-draw data
+         *
+         * For example, a non-animated mesh just needs a mat4 for its model matrix
+         *
+         * This buffer gets re-written to every frame, since the number of renderables in this mesh batch might have changed. If there's
+         * more renderables than the buffer can hold, it gets reallocated from the RHI
+         */
+        rhi::Buffer* per_renderable_data = nullptr;
+
+        std::vector<RenderCommandType> commands;
+
+        ProceduralMeshBatch(std::unordered_map<MeshId, ProceduralMesh>& meshes, MeshId key) : mesh(meshes, key) {}
```

Comment:
Does MapAccessor store a reference? If so this is valid code, if not, you should be taking meshes by value and moving.

Reply:
Answered my own question, ignore.